Userland/DevTools/UserspaceEmulator/SoftCPU.cpp

```diff
@@ -1665,7 +1682,12 @@ void SoftCPU::FIADD_RM32(const X86::Instruction& insn)
     fpu_set(0, fpu_get(0) + (long double)m32int);
 }
 
-void SoftCPU::FCMOVB(const X86::Instruction&) { TODO_INSN(); }
+void SoftCPU::FCMOVB(const X86::Instruction& insn)
+{
+    VERIFY(insn.modrm().is_register());
+    if (cf())
+        fpu_set(0, fpu_get(insn.rm() & 7)); // why is this the way to get "the second operand"?
```

Comment:
Please don't ask questions by adding comments to the code. Ask a person instead :)

This is currently the way we extract the second operand for FPU instructions because nobody invented a nicer abstraction. We talked about it when @nico added the initial FPU support, but nothing came of it.

It could be nice to add some sort of helper to `X86::Instruction` for this :)

Reply:
Ok I will remove that for now, Am working on more FPU stuff at the moment anyway, so I probably gonna change that in the nearer future